lib/libzfs/os/linux/libzfs_util_os.c

```diff
@@ -76,93 +80,74 @@ libzfs_error_init(int error)
 	}
 }
 
-static int
-libzfs_module_loaded(const char *module)
-{
-	const char path_prefix[] = "/sys/module/";
-	char path[256];
-
-	memcpy(path, path_prefix, sizeof (path_prefix) - 1);
-	strcpy(path + sizeof (path_prefix) - 1, module);
-
-	return (access(path, F_OK) == 0);
-}
-
 /*
- * Verify the required ZFS_DEV device is available and optionally attempt
- * to load the ZFS modules.  Under normal circumstances the modules
- * should already have been loaded by some external mechanism.
+ * zfs(4) is loaded by udev if there's a fstype=zfs device present,
+ * but if there isn't, load them automatically;
+ * always wait for ZFS_DEV to appear via udev.
  *
  * Environment variables:
- * - ZFS_MODULE_LOADING="YES|yes|ON|on" - Attempt to load modules.
```

Comment:
I'm fine with changing the default behavior, that's overdue, but let's retain the `ZFS_MODULE_LOADING` environment variable.  I can imagine environments where it's preferable the command immediately fail instead of blocking for 10 seconds (or best case 1 second).

We should also add these to the `zpool(8)` man page.

Reply:
Re-adding `ZFS_MODULE_LOADING` would do nothing for timeouts, that's `ZFS_MODULE_TIMEOUT=0`. The only thing having `ZFS_MODULE_LOADING=false` would change would be running `modprobe zfs`, which is instant.

Reply:
Re-adding `ZFS_MODULE_LOADING` would do nothing for timeouts, that's `ZFS_MODULE_TIMEOUT=0`. The only thing having `ZFS_MODULE_LOADING=false` would change would be running `modprobe zfs`, which is instant.

I rolled back `ZFS_MODULE_TIMEOUT=0` to being "give up instantly", "never give up" is now <0, and documented this in zfs.8 and zpool.8

Reply:
That works for me.